File: lista-de-exercicios/exercicio-9/NovoFibonacci.test.ts
import { jest } from '@jest/globals';
import { NovoFibonacci } from './NovoFibonacci';

describe('Classe NovoFibonacci', () => {
    test('Deve retornar um array onde o último número é menor ou igual ao número informado', () => {
        const novoFibonacci0 = new NovoFibonacci(0);
        const novoFibonacci1 = new NovoFibonacci(1);
        const novoFibonacci2 = new NovoFibonacci(2);

        expect(novoFibonacci0.contar()).toStrictEqual([0]);
        expect(novoFibonacci1.contar()).toStrictEqual([0, 1]);
        expect(novoFibonacci2.contar()).toStrictEqual([0, 1, 1, 2]);
    })

    test('Deve retornar um erro se entrar um número decimal ou negativo', () => {
        const novoFibonacciNegativo = new NovoFibonacci(-1);
        const novoFibonacciDecimal = new NovoFibonacci(1.2);
        const novoFibonacciNaN = new NovoFibonacci(NaN);

        expect(() => novoFibonacciNegativo.contar()).toThrowError();
        expect(() => novoFibonacciDecimal.contar()).toThrowError();
        expect(() => novoFibonacciNaN.contar()).toThrowError();
    })
})

File: lista-de-exercicios-java/src/test/java/org/example/exercicio09/NovoFibonacciTest.java
package org.example.exercicio09;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;

import static org.junit.jupiter.api.Assertions.*;

class NovoFibonacciTest {
    @DisplayName("Deve retornar um array onde o último número é menor ou igual ao número informado")
    @Test
    void deveRetornarArrayAteNumeroInformadoOuMenor() {
        NovoFibonacci novoFibo = new NovoFibonacci();
        ArrayList<Integer> novoFiboArray = novoFibo.serializar(2);

        assertEquals(4, novoFiboArray.size());
    }

    @DisplayName("Deve retornar o array correto")
    @Test
    void deveRetornarArrayCorreto() {
        NovoFibonacci novoFibo = new NovoFibonacci();

        assertArrayEquals(new Integer[] {0, 1, 1, 2, 3, 5, 8, 13, 21}, novoFibo.serializar(25).toArray());
    }

    @DisplayName("Deve retornar um erro se entrar um número negativo")
    @Test
    void deveRetornarErroComNumeroNegativo() {
        NovoFibonacci novoFibo = new NovoFibonacci();

        assertThrows(IllegalArgumentException.class, () -> novoFibo.serializar(-2));
    }
}